### PR TITLE
Worldstate-only resync behavior

### PR DESCRIPTION
**Pull request by garyschulte**
## PR description
Implements a  debug rpc endpoint and a behavior to resync just the worldstate.  

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Comment by garyschulte** [comment]
@shemnon as far as UX for a resync feature, 'touching' a file in the data dir is rather simple to implement and accessible.  A debug rpc endpoint may or may not be a good idea for this kind of functionality.  What do you think about file vs rpc, file and rpc, or some other command pattern to trigger this behavior?

**Comment by shemnon** [comment]
I'm not a devops maven, I'd think we should get the UX feedback from someone who manages fleets of nodes.  Touching a file feels hacky to me but I've discovered lots of DevOps "best practices" feel hacky to me.

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java [hunk 1]
```diff
@@ -34,6 +34,7 @@ public enum RpcMethod {
   CLIQUE_GET_SIGNER_METRICS("clique_getSignerMetrics"),
   DEBUG_ACCOUNT_AT("debug_accountAt"),
   DEBUG_METRICS("debug_metrics"),
+  DEBUG_RESYNC_WORLDSTATE("debug_resyncWorldState"),
```
Comment (matkt) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java:37]: I will prefer to have this king of RPC in the admin namespace

**Review by matkt — CHANGES_REQUESTED** [review]

**Comment by matkt** [comment]
> I'm not a devops maven, I'd think we should get the UX feedback from someone who manages fleets of nodes. Touching a file feels hacky to me but I've discovered lots of DevOps "best practices" feel hacky to me.

I'm no devops but I personally prefer an RPC call or a flag to a file that feels DIY.
I would just say that the RPC call must be admin because it seems too critical to me to simply be in debug

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/BadBlockManager.java [hunk 2]
```diff
@@ -49,6 +49,12 @@ public void addBadBlock(final Block badBlock, final Optional<Throwable> cause) {
     }
   }
 
+  public void reset() {
+    this.badBlocks.invalidateAll();
+    this.badHeaders.invalidateAll();
+    this.latestValidHashes.invalidateAll();
+  }
+
```
Comment (fab-10) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/BadBlockManager.java:57]: Where is this called?

**Review by fab-10 — APPROVED** [review]
LGTM, just a method that seems unused

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/BadBlockManager.java (hunk 2 above)
Comment (garyschulte) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/BadBlockManager.java:57]: https://github.com/hyperledger/besu/pull/4875/files#diff-cc598aa8b256f675dfcbd331db0955fd273916856708992c92364aa373405f12R49

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java (hunk 1 above)
Comment (garyschulte) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java:37]: as discussed, we will keep this in debug namespace

**Pull request merged**